pkg/pubsub/adapter/adapter.go

```diff
@@ -18,253 +18,202 @@ package adapter
-	//  for multiple source objects. Same with Namespace, when doing multi-tenancy.
 	args := &ReportArgs{
-		Name:          a.Name,
-		Namespace:     a.Namespace,
-		EventType:     event.Type(),
-		EventSource:   event.Source(),
-		ResourceGroup: a.ResourceGroup,
+		EventType:   event.Type(),
+		EventSource: event.Source(),
 	}
 
-	var err error
+	message := (*binding.EventMessage)(event)
+
 	// If a transformer has been configured, then transform the message.
 	// Note that this path in the code will be executed when using the receive adapter as part of the underlying Channel
 	// of a Broker. We currently set the TransformerURI to be the address of the Broker filter pod.
-	// TODO consider renaming transformer as it is confusing.
-	if a.transformer != nil {
-		transformedCTX, transformedEvent, err := a.transformer.Send(ctx, event)
-		rtctx := cloudevents.HTTPTransportContextFrom(transformedCTX)
+	// TODO consider renaming transformer as it is confusing. And see if we can get rid of this.
+	if a.transformerURI != "" {
+		resp, err := a.sendMsg(ctx, a.transformerURI, message)
 		if err != nil {
-			logger.Errorf("error transforming cloud event %q", event.ID())
-			a.reporter.ReportEventCount(args, rtctx.StatusCode)
-			return err
+			a.logger.Error("Failed to send message to transformer", zap.String("address", a.transformerURI), zap.Error(err))
+			msg.Nack()
+			return
+		}
+
+		defer func() {
+			if err := resp.Body.Close(); err != nil {
+				a.logger.Warn("Failed to close response body", zap.Error(err))
+			}
+		}()
+
+		a.reporter.ReportEventCount(args, resp.StatusCode)
+
+		if resp.StatusCode/100 != 2 {
+			a.logger.Error("Event delivery failed", zap.Int("StatusCode", resp.StatusCode))
+			msg.Nack()
+			return
 		}
-		if transformedEvent == nil {
+
+		respMsg := cehttp.NewMessageFromHttpResponse(resp)
+		if respMsg.ReadEncoding() == binding.EncodingUnknown {
 			// This doesn't mean there was an error. E.g., the Broker filter pod might not return a response.
-			// Report the returned Status Code and return.
-			logger.Debugf("cloud event %q was not transformed", event.ID())
-			a.reporter.ReportEventCount(args, rtctx.StatusCode)
-			return nil
+			msg.Ack()
+			return
 		}
-		// Update the event with the transformed one.
-		event = *transformedEvent
+
+		e, err := binding.ToEvent(ctx, respMsg)
+		if err != nil {
+			a.logger.Error("Failed to convert response message to event",
+				zap.Any("response", respMsg), zap.Error(err))
+			msg.Ack()
+			return
+		}
+
+		// TODO check if this is OK
 		// Update the tracing information to use the span returned by the transformer.
-		ctx = trace.NewContext(ctx, trace.FromContext(transformedCTX))
-	}
+		// ctx = trace.NewContext(ctx, trace.FromContext(transformedCTX))
+		if span := trace.FromContext(ctx); span.IsRecordingEvents() {
+			span.Annotate(ceclient.EventTraceAttributes(e),
+				"Event reply",
+			)
+		}
```

Comment:
@ian-mi @Harwayne 
here is the part of tracing that I have no clue about. Is this somewhat equivalent to the old:
`trace.NewContext(ctx, trace.FromContext(transformedCTX))` ?

Reply:
OK, seems that the e2e tests failing are due to tracing problems. I'm surely doing something wrong...